front/components/providers/AnthropicSetup.tsx

```diff
@@ -0,0 +1,203 @@
+import { ActionButton, Button } from "@app/components/Button";
+import { checkProvider } from "@app/lib/providers";
+import { Dialog, Transition } from "@headlessui/react";
+import { Fragment, useEffect, useState } from "react";
+import { useSWRConfig } from "swr";
+
+type AnthropicSetupProps = {
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  config: { [key: string]: string };
+  enabled: boolean;
+};
+
+export default function AnthropicSetup({
```

Comment:
There's a lot of repetition between the provider setup components -- Might be nice to factorize them a bit at some point, WDYT @spolu @lasryaric  ?

Reply:
Definitely. We can do that after the move to Workspace :+1: